### PR TITLE
[NewIR] enable new ir for unittests in white list

### PR DESCRIPTION
**Pull request by kangguangli**
### PR types

New features
### PR changes

Others
### Description

This PR is used to enable new ir for unittests in white list. If someone want to add a unittest, just add it in `test/new_ir_op_test_white_list`.

#### Details
If an op need to be executed in new ir, two flags needs to be set:
1. `FLAGS_NEW_IR_OPTEST`, now set in `paddle/scripts/paddle_build.sh`
2. `FLAGS_NEW_IR_OPTEST_WHITE_LIST`, now set in `test/legacy_test/CMakeLists.txt`

#### Future Works
Due to the constraints of cmake, we can only set the test property in the scope it defined. So if we want to add the test not in `test/legacy_test/`, we need to modify the corresponding `CMakeLists`.

#### Others
Pcard-67164

### CODE REVIEW

File: test/legacy_test/eager_op_test.py [hunk 1]
```diff
@@ -1125,6 +1127,41 @@ def _calc_dygraph_output(
             )
             return outputs
 
+    def _check_ir_output(self, place, program, feed_map, fetch_list, outs):
+        if os.getenv("FLAGS_NEW_IR_OPTEST") is None:
```
Comment (Charles-hit) [inline test/legacy_test/eager_op_test.py:1131]: 这儿是不是可以参考一下以前optest的白名单机制

Comment (kangguangli) [reply test/legacy_test/eager_op_test.py:1131]: 这里需求不太一样，之前OpTest的白名单机制不支持单测粒度的白名单

**Review by risemeup1 — APPROVED** [review]
LGTM

**Review by XieYunshen — APPROVED** [review]

**Review by Aurelius84 — APPROVED** [review]
LGTM

**Review by Charles-hit — APPROVED** [review]

**Pull request merged**